src/openshift/component.ts

```diff
@@ -14,36 +14,76 @@ import { Url } from './url';
 import { CliExitData } from '../cli';
 
 export class Component extends OpenShiftItem {
+
     static async create(application: OpenShiftObject): Promise<string> {
-        // should use QuickPickItem with label and description
-        const sourceTypes: vscode.QuickPickItem[] = [
-            {
-                label: 'Git Repository',
-                description: 'Use an existing git repository as a source for the component'
-            },
-            {
-                label: 'Binary File',
-                description: 'Use binary file as a source for the component'
-            },
-            {
-                label: 'Workspace Directory',
-                description: 'Use workspace directory as a source for the component'
+        const data: OpenShiftObject = await Component.getApplicationData(application);
+        if (data) {
+            const sourceTypes: vscode.QuickPickItem[] = [
+                {
+                    label: 'Git Repository',
+                    description: 'Use an existing git repository as a source for the component'
+                },
+                {
+                    label: 'Binary File',
+                    description: 'Use binary file as a source for the component'
+                },
+                {
+                    label: 'Workspace Directory',
+                    description: 'Use workspace directory as a source for the component'
+                }
+            ];
+            const componentSource = await vscode.window.showQuickPick(sourceTypes, {
+                placeHolder: "Select source type for component"
+            });
+            if (!componentSource) return null;
+
+            let command: Promise<string>;
+            if (componentSource.label === 'Git Repository') {
+                command = Component.createFromGit(data);
+            } else if (componentSource.label === 'Binary File') {
+                command = Component.createFromBinary(data);
+            } else {
+                command = Component.createFromLocal(data);
             }
-        ];
-        const componentSource = await vscode.window.showQuickPick(sourceTypes, {
-            placeHolder: "Select source type for component"
-        });
-        if (!componentSource) return null;
+            return command.catch((err) => Promise.reject(`Failed to create component with error '${err}'`));
+        }
+    }
 
-        let command: Promise<string>;
-        if (componentSource.label === 'Git Repository') {
-            command = Component.createFromGit(application);
-        } else if (componentSource.label === 'Binary File') {
-            command = Component.createFromBinary(application);
-        } else {
-            command = Component.createFromLocal(application);
+    static async getApplicationData(application) {
```

Comment:
Please see https://github.com/redhat-developer/vscode-openshift-tools/pull/432#discussion_r245595571 and #499 for the summary of my issues with this implementation.